GitHubReactions.ps1

```diff
@@ -0,0 +1,493 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+@{
+    GitHubReactionTypeName = 'GitHub.Reaction'
+ }.GetEnumerator() | ForEach-Object {
+     Set-Variable -Scope Script -Option ReadOnly -Name $_.Key -Value $_.Value
+ }
+
+filter Get-GitHubReaction
+{
+<#
+    .SYNOPSIS
+        Retrieve reactions of a given GitHub issue.
+
+    .DESCRIPTION
+        Retrieve reactions of a given GitHub issue.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OwnerName
+        Owner of the repository.
+        If not supplied here, the DefaultOwnerName configuration property value will be used.
+
+    .PARAMETER RepositoryName
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER IssueNumber
+        The issue *OR* pull request number. This parameter has an alias called
+        `PullRequestNumber` so that the experience feels natural but in this case,
+        PRs are handled the same as issues.
+
+    .PARAMETER ReactionType
+        The type of reaction you want to retrieve. This is also called the 'content' in the GitHub API.
+        Valid options are based off: https://developer.github.com/v3/reactions/#reaction-types
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .INPUTS
```

Comment:
So I've made this change... although I question the use. It feels like `.INPUTS` should only contain things that perfectly compose.

Here's an example:

In `Get-GitHubReaction` I have `GitHub.IssueComment` as an input because it provides a `RepositoryUrl` and `IssueNumber`... however... that will react to the _Issue_ and not the _IssueComment_. That makes this behavior unexpected IMO.

I want to put `GitHub.IssueComment` when it's properly supported... but I left it in for now to be consistent.

Reply:
I get that.  There's a solid argument for that.  And yet, there's still value from someone piping in any of those other objects because they do indicate context for what they're looking for.   I think for now, we leave it the way you've done it, and we can revisit in later PR's if there's feedback that doing it this way really wasn't a good idea afterall.